openfeature/api.go

```diff
@@ -49,7 +49,31 @@ func (api *evaluationAPI) setProvider(provider FeatureProvider) error {
 	oldProvider := api.defaultProvider
 	api.defaultProvider = provider
 
-	api.initNewAndShutdownOld(provider, oldProvider)
+	api.initNewAndShutdownOldAsync(provider, oldProvider)
+	err := api.eventExecutor.registerDefaultProvider(provider)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// setProviderAndWait sets the default FeatureProvider of the evaluationAPI. Returns an error if FeatureProvider is nil
+// This is a blocking call and will wait for the provider to be ready
+func (api *evaluationAPI) setProviderAndWait(provider FeatureProvider) error {
```

Comment:
Nitpick: to stay consistent with the naming i would suggest to call this function `setProviderSync`

Reply:
Agreed. The spec doesn't dictate function names. I think your suggestion makes sense here.